### DISCUSSION TITLE
Attachment Field

### DISCUSSION THREAD

**Discussion by ikaheraJQ** [Budibase/budibase#6242] [Help] [answered]
Excuse me, I want to ask about attachment field.
I tried to assign Attachment Field value to Text Field and I got an array like this,
[{"size":839884,"name":"xxx.pdf","url":"/prod-budi-app-assets/app_default_ca5976dc37xc73866abf2103/attachments/5cb1fdad-1c11-47a2-97a9-32f2e28f7252.pdf","extension":"pdf","key":"app_default_ca5976dc370442af9aec738x/attachments/5cb1fdad-1c11-47a2-97a9-32f2e28f7252.pdf"}]
My question is if I don't store the attachment into Budibase DB, how long the uploaded file url from that array will be exist?
Thank you 🙏🏽

**Comment by melohagan** [comment, marked as answer]
Hi @ikaheraJQ 

Currently, the uploaded files will remain permanently unless manually removed from MinIO manually: https://github.com/Budibase/budibase/issues/5564

As for the URLs, they are created when a value is assigned to an attachment field, and I imagine they are deleted when the attachment field is cleared.

What is your thought process - why is the file url important to you?

**Reply by ikaheraJQ** [reply]
Actually, I need to create a feature for uploading file. The file needs to be stored in external DB and the user can download it again by clicking a button or icon or link. However I can't find a way to convert the file to base64 in Budibase. So I think I will store the uploaded file url instead of the file so the user can download it again. Or do you have any suggestion for this case?

**Reply by melohagan** [reply]
There is a feature request open for supporting external DB attachments which would be great in this case: https://github.com/Budibase/budibase/issues/4302

Related discussion on base64 image files, but I wasn't of much help: https://github.com/Budibase/budibase/discussions/5766

So I do think the best bet is to store the uploaded file url.

**Reply by ikaheraJQ** [reply]
Okay, thank you for the suggestion 🙏🏽
